### ISSUE TITLE
 Unable to load package support for: sympc. No module named 'sympc'

### ISSUE THREAD

**Issue by spnzig** [OpenMined/PySyft#5671]
Trying out examples for SMC with PySyft, but I am getting errors on loading "sympc"

```
import torch  
import syft as sy  

sy.load("sympc") 
sy.logger.add(sink="./example.log")
```

![image](https://user-images.githubusercontent.com/30037676/122505490-141b6980-d01a-11eb-9cdb-2d20bbd68a15.png)

## System Information
 - OS: Windows 10
 - Language Version: Python 3.7
 - Pysyft version : 0.5.0rc2

**Comment by rasswanth-s** [comment]
@spnzig  ,did you try installing sympc ,`pip install git+https://github.com/OpenMined/SyMPC@dev` and checking it again.

**Comment by spnzig** [comment]
@rasswanth-s yes I tried that. It still didn't work. 
![image](https://user-images.githubusercontent.com/30037676/122704993-766ea700-d272-11eb-8b63-df4d198b146c.png)

**Comment by rasswanth-s** [comment]
Sorry ,I interchanged in pysyft we dev for SyMPC we use main , this would work pip install git+https://github.com/OpenMined/SyMPC@main

**Comment by spnzig** [comment]
@rasswanth-s  its gives the below error. 
![image](https://user-images.githubusercontent.com/30037676/122705334-204e3380-d273-11eb-9593-8f0426bcf0c4.png)

**Comment by rasswanth-s** [comment]
I think `sy.load`  is deprecated now, could you try `import sympc` in a separate cell.

**Comment by spnzig** [comment]
Oh, I see.
@rasswanth-s that doesn't seem to work for me either . 
![image](https://user-images.githubusercontent.com/30037676/122706601-f0545f80-d275-11eb-8b0f-50154a7eb127.png)

Is there any updated document with details about the installation steps? 
I am trying out the examples for SyMPC under PySyft. 
![image](https://user-images.githubusercontent.com/30037676/122706822-5c36c800-d276-11eb-9e87-153e9cf7cd44.png)

**Comment by rasswanth-s** [comment]
Yeah , that is due to gpu build issue,I think you must have installed `torchcsprng==0.2.1`.
Firtst  you  could uninstall it `pip uninstall torchcsprng`.
Then install the cpu build from torch repo `pip install torchcsprng==0.2.1+cpu -f https://download.pytorch.org/whl/torch_stable.html`
Then you could import sympc and check.

**Comment by spnzig** [comment]
Thank you. I will try that out. 
I have installed PySyft with this command ` pip install syft==0.5.0rc2` and haven't installed any other package explicitly to try out the examples for SMC with PySyft.
Would this have any dependency issues for running other examples under Homomorphic encryption or differential privacy?

**Comment by rasswanth-s** [comment]
It is due to the PyPI problem, now it should work. I think there  wont be any dependency issue with other libraries ,you could open a issue , if there are any.

**Comment by spnzig** [comment]
Thanks @rasswanth-s . That worked. But I am not sure if the library is fully functional. I get errors when I try out the code.
![image](https://user-images.githubusercontent.com/30037676/122738589-c0ba4d00-d29f-11eb-9c21-e81cff60b2b4.png)

The error is below
```
`[2021-06-21T14:49:49.943489+0530][CRITICAL][logger]][10676] UnknownPrivateException has been triggered.
---------------------------------------------------------------------------
UnknownPrivateException                   Traceback (most recent call last)
<ipython-input-13-b434b26d9c89> in <module>
----> 1 print("X + Y =", (x + y).reconstruct())

E:\SMC\lib\site-packages\sympc\tensor\mpc_tensor.py in reconstruct(self, decode, get_shares)
    375             self.share_ptrs,
    376             get_shares=get_shares,
--> 377             security_type=self.session.protocol.security_type,
    378         )
    379 

E:\SMC\lib\site-packages\sympc\tensor\share_tensor.py in reconstruct(share_ptrs, get_shares, security_type)
    525 
    526         args = [[share] for share in share_ptrs]
--> 527         local_shares = request_wrap(args)
    528 
    529         shares = [share.tensor for share in local_shares]

E:\SMC\lib\site-packages\sympc\utils\utils.py in wrapper(args, kwargs)
    122                 futures.append(executor.submit(funcs[i], *_args, **_kwargs))
    123 
--> 124         local_shares = [f.result() for f in futures]
    125 
    126         return local_shares

E:\SMC\lib\site-packages\sympc\utils\utils.py in <listcomp>(.0)
    122                 futures.append(executor.submit(funcs[i], *_args, **_kwargs))
    123 
--> 124         local_shares = [f.result() for f in futures]
    125 
    126         return local_shares

C:\ProgramData\Anaconda3\lib\concurrent\futures\_base.py in result(self, timeout)
    423                 raise CancelledError()
    424             elif self._state == FINISHED:
--> 425                 return self.__get_result()
    426 
    427             self._condition.wait(timeout)

C:\ProgramData\Anaconda3\lib\concurrent\futures\_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

C:\ProgramData\Anaconda3\lib\concurrent\futures\thread.py in run(self)
     55 
     56         try:
---> 57             result = self.fn(*self.args, **self.kwargs)
     58         except BaseException as exc:
     59             self.future.set_exception(exc)

E:\SMC\lib\site-packages\sympc\tensor\share_tensor.py in _request_and_get(share_ptr)
    518             if not islocal(share_ptr):
    519                 share_ptr.request(block=True)
--> 520             res = share_ptr.get_copy()
    521             return res
    522 

e:\smc\src\syft\packages\syft\src\syft\core\pointer\pointer.py in get_copy(self, request_block, timeout_secs, reason, verbose)
    202             reason=reason,
    203             delete_obj=False,
--> 204             verbose=verbose,
    205         )
    206 

e:\smc\src\syft\packages\syft\src\syft\core\pointer\pointer.py in get(self, request_block, timeout_secs, reason, delete_obj, verbose)
    267 
    268         if not request_block:
--> 269             result = self._get(delete_obj=delete_obj, verbose=verbose)
    270         else:
    271             response_status = self.request(

e:\smc\src\syft\packages\syft\src\syft\core\pointer\pointer.py in _get(self, delete_obj, verbose)
    177         )
    178 
--> 179         obj = self.client.send_immediate_msg_with_reply(msg=obj_msg).data
    180         if self.is_enum:
    181             enum_class = self.client.lib_ast.query(self.path_and_name).object_ref

e:\smc\src\syft\packages\syft\src\syft\core\node\common\client.py in send_immediate_msg_with_reply(self, msg, route_index)
    234                 exception = exception_msg.exception_type(exception_msg.exception_msg)
    235                 error(str(exception))
--> 236                 traceback_and_raise(exception)
    237             else:
    238                 return response.message

e:\smc\src\syft\packages\syft\src\syft\logger.py in traceback_and_raise(e, verbose)
     59     if not issubclass(type(e), Exception):
     60         e = Exception(e)
---> 61     raise e
     62 
     63 

UnknownPrivateException: UnknownPrivateException has been triggered.`
```

**Comment by rasswanth-s** [comment]
I think we did not update the examples to the current sympc framework, I would look at it and get back to you regarding this.

**Comment by spnzig** [comment]
@rasswanth-s Thanks for your time. That would be really helpful.

**Comment by spnzig** [comment]
@rasswanth-s any suggestion on how to go ahead with the MPC examples? I still seem to be stuck with an Unknown private exception when I try basic operations (sum, subtract, multiply) on tensors.
![image](https://user-images.githubusercontent.com/30037676/124871533-c7f09300-dfe1-11eb-91f7-451d5b9779b1.png)

**Comment by rasswanth-s** [comment]
The MPC Examples in PySyft are not updated, you could try the MPC examples in SyMPC repo,(https://github.com/OpenMined/SyMPC/tree/main/examples) ,until PySyft examples are fixed.

**Comment by madhavajay** [comment]
@spnzig We have a new code architecture in dev which will be released as `0.7.0` soon with a much better working SMPC implementation built into the core.
